src/app/drive/components/ShareDialog/ShareDialog.tsx

```diff
@@ -151,7 +153,7 @@ const ShareDialog = (props: ShareDialogProps): JSX.Element => {
   const userList = useRef<HTMLDivElement>(null);
   const userOptions = useRef<HTMLButtonElement>(null);
 
-  const itemOwnerEmail = props?.isDriveItem ? '' : (itemToShare?.item as AdvancedSharedItem).user?.email;
+  const itemOwnerEmail = props?.isDriveItem ? '' : (itemToShare?.item as AdvancedSharedItem)?.user?.email;
   const isUserOwner = (!!itemOwnerEmail && itemOwnerEmail === props?.user?.email) || !!props?.isDriveItem;
```

Comment:
`AdvancedShareUtils.isUserOwner()` might by  a better way to handle this, so you have your logic centralized in one way, and every component that uses this, uses the same centralized logic